File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        int n = nums.size();
        map<int,int>map;
        for(int i=0;i<n;i++){
            map[nums[i]]++;
        }

        vector<int>ans;
        vector<vector<int>>pair;
        for(auto i:map){
            vector<int>pop;
            pop.push_back(i.second);
            pop.push_back(i.first);
            pair.push_back(pop);
        }


        sort(pair.begin(),pair.end());

        for(auto i:pair){
            cout<<i[0]<<" "<<i[1]<<endl;
        }

        for(int i=pair.size()-1;i>=0;i--){
            for(int j=i-1;j>=0;j--){
                if(pair[i][0]==pair[j][0]){
                    if(pair[i][1]>pair[j][1]){
                        int p=pair[j][1];
                        pair[j][1] = pair[i][1];
                        pair[i][1]=p;
                    }
                
                }
            }
        }

        for(auto i:pair){
            int t=i[0];
            while(t>0){
                ans.push_back(i[1]);
                t--;
            }
        }

        return ans;
    }
};

File: 1741-sort-array-by-increasing-frequency/1741-sort-array-by-increasing-frequency.java
class Solution {
    class Pair{
        int num;
        int freq;
        Pair(int num, int freq){
            this.num=num;
            this.freq=freq;
        }
    }
    public int[] frequencySort(int[] nums) {
        HashMap<Integer, Integer>h=new HashMap<>();
        for(int i=0; i<nums.length; i++){
            h.put(nums[i], h.getOrDefault(nums[i],0)+1);
        }
        PriorityQueue<Pair>pq=new PriorityQueue<>((a,b)->a.freq==b.freq?b.num-a.num: a.freq-b.freq);
        for(int i: h.keySet()){
            pq.add(new Pair(i, h.get(i)));
        }
        int arr[]=new int[nums.length];
        int j=0;
        while(!pq.isEmpty()){
            int n=pq.peek().num;
            int f=pq.peek().freq;
            pq.poll();
            for(int i=0; i<f; i++){
                arr[j]=n;
                j++;
            }
        }
        return arr;
    }
}